easybuild/tools/filetools.py

```diff
@@ -1150,6 +1151,35 @@ def rmtree2(path, n=3):
     else:
         _log.info("Path %s successfully removed." % path)
 
+def find_backup_name_candidate(src_file):
+    """Returns a non-existing file to be used as destination for rotating backups"""
+    dst_file = src_file
+
+    if os.path.exists(src_file):
+        i = 0
+        dst_file = "%s_%d" % (src_file, i)
+        while os.path.exists(dst_file):
+            i += 1
+            dst_file = "%s_%d" % (src_file, i)
+
+    return dst_file
+
+
+def back_up_file(src_file, backup_extension=""):
+    """Backs up a file appending a backup extension and a number to it. Returns the name of the backup"""
```

Comment:
clarify: the number is optional (only when existing backups are already there)

use `:param:` to document arguments

also support `hide` named argument to backup to a hidden file